core/src/block.rs

```diff
@@ -146,59 +149,40 @@ impl<'x> OpenBlock<'x> {
         Ok(r)
     }
 
-    /// Push a transaction into the block.
-    pub fn push_transaction<C: FindActionHandler>(
-        &mut self,
-        tx: VerifiedTransaction,
-        h: Option<TxHash>,
-        client: &C,
-        parent_block_number: BlockNumber,
-        parent_block_timestamp: u64,
-    ) -> Result<(), Error> {
-        if self.block.transactions_set.contains(&tx.hash()) {
-            return Err(HistoryError::TransactionAlreadyImported.into())
-        }
+    pub fn begin_transaction_execution(&mut self, coordinator: &Coordinator, evidences: Vec<Evidence>) {
```

Comment:
Why is it named `begin_transaction_execution` instead of `open_block`?
I think the current name can confuse users to believe it should be called before every `execute_transactions`.

Reply:
I thought `open_block.open_block()` was an confusing naming.
I agree that the current name is also confusing as well.
Could you suggest any better name?
Or I can merge the method into `OpenBlock::try_new`, but in that case, the `try_new` method will become complicated.